nbgrader/apps/fetchapp.py

```diff
@@ -73,16 +73,19 @@ def _load_config(self, cfg, **kwargs):
     def start(self):
         super(FetchApp, self).start()
 
-        # set assignemnt and course
-        if len(self.extra_args) == 1:
-            self.coursedir.assignment_id = self.extra_args[0]
-        elif len(self.extra_args) > 2:
-            self.fail("Too many arguments")
-        elif self.coursedir.assignment_id == "":
+        # set assignment and course
+        if len(self.extra_args) == 0:
```

Comment:
This is causing the tests to fail, because now if `--assignment` is passed (and no positional arguments), it won't work. So we need to check for `if len(self.extra_args) == 0 and self.coursedir.assignment_id == ""` and handle the case where the assignment id is set.